base/tks/src/main/java/org/dogtagpki/server/tks/servlet/TokenServlet.java

```diff
@@ -1346,12 +1387,31 @@ private void processComputeSessionKey(HttpServletRequest req,
         }
 
         if (serversideKeygen == true) {
-            if (drm_trans_wrapped_desKey != null && drm_trans_wrapped_desKey.length > 0)
+            if (drm_trans_wrapped_desKey != null && drm_trans_wrapped_desKey.length > 0) {
                 drm_trans_wrapped_desKeyString =
                         org.mozilla.jss.netscape.security.util.Utils.SpecialEncode(drm_trans_wrapped_desKey);
-            else {
+            } else {
+                status = "1";
+            }
+
+            if (drm_trans_wrapped_aesKey != null &&  drm_trans_wrapped_aesKey.length > 0) {
+                drm_trans_wrapped_aesKeyString =
+                        org.mozilla.jss.netscape.security.util.Utils.SpecialEncode(drm_trans_wrapped_aesKey);
+                logger.debug("drm_trans_wrapped_aesKeyString: " + drm_trans_wrapped_aesKeyString);
+
+            } else {
                 status = "1";
             }
+
+            if (aes_wrapped_desKey != null &&  aes_wrapped_desKey.length > 0) {
+                aes_wrapped_desKeyString =
```

Comment:
shouldn't you change status back to "0" here?  But again, if this is "either des or aes" there's probably better way to separate the code.

Reply:
Once again this is in processComputeSessionKey for scp01. Will deal with it.